src/math/curve-iterator.js

```diff
@@ -0,0 +1,225 @@
+                    this.recip = (isFinite(diff) ? diff : 0);
+                    this.p0 = keys[index][1];
+                    this.p1 = keys[index + 1][1];
+                    if (this._isHermite()) {
+                        this._calcTangents(keys, index);
+                    }
+                }
+            }
+        },
+
+        // returns true if the curve is a hermite and false otherwise
+        _isHermite: function () {
+            return [pc.CURVE_CATMULL,
+                pc.CURVE_CARDINAL,
+                pc.CURVE_CARDINAL_STABLE].indexOf(this.curve.type) != -1;
+        },
+
+        // calculate tangents for the hermite curve
+        _calcTangents: function (keys, index) {
+            var a;
+            var b = keys[index];
+            var c = keys[index + 1];
+            var d;
+
+            if (index === 0) {
+                a = [keys[0][0] + (keys[0][0] - keys[1][0]),
+                    keys[0][1] + (keys[0][1] - keys[1][1])];
+            } else {
+                a = keys[index - 1];
+            }
+
+            if (index == keys.length - 2) {
+                d = [keys[index + 1][0] + (keys[index + 1][0] - keys[index][0]),
+                    keys[index + 1][1] + (keys[index + 1][1] - keys[index][1])];
+            } else {
+                d = keys[index + 2];
+            }
+
+            if (this.curve.type === pc.CURVE_CARDINAL_STABLE) {
+                // calculate tangent scale (due to non-uniform knot spacing)
+                var s1_ = 2 * (c[0] - b[0]) / (c[0] - a[0]);
+                var s2_ = 2 * (c[0] - b[0]) / (d[0] - b[0]);
+
+                this.m0 = this.curve.tension * (isFinite(s1_) ? s1_ : 0) * (c[1] - a[1]);
+                this.m1 = this.curve.tension * (isFinite(s2_) ? s2_ : 0) * (d[1] - b[1]);
+            } else {
+                // original tangent scale calc
+                var s1 = (c[0] - b[0]) / (b[0] - a[0]);
+                var s2 = (c[0] - b[0]) / (d[0] - c[0]);
+
+                var a_ = b[1] + (a[1] - b[1]) * (isFinite(s1) ? s1 : 0);
+                var d_ = c[1] + (d[1] - c[1]) * (isFinite(s2) ? s2 : 0);
+
+                var tension = (this.curve.type === pc.CURVE_CATMULL) ? 0.5 : this.curve.tension;
+
+                this.m0 = tension * (c[1] - a_);
+                this.m1 = tension * (d_ - b[1]);
+            }
+        },
+
+        _evaluateHermite: function (p0, p1, m0, m1, t) {
+            var t2 = t * t;
+            var twot = t + t;
+            var omt = 1 - t;
+            var omt2 = omt * omt;
+            return p0 * ((1 + twot) * omt2) +
+                   m0 * (t * omt2) +
+                   p1 * (t2 * (3 - twot)) +
+                   m1 * (t2 * (t - 1));
+        },
+
+        /**
+         * @function
+         * @name pc.CurveIterator#time
+         * @description Get the current iterator time
+         * @returns {Number} The current time.
+         */
+        time: function () {
```

Comment:
Why isn't this a vanilla property?

Reply:
The time property is actually internal state and should only be set by the functions evaluate() and advance(), never by the caller. I guess this accessor is an attempt to communicate that. I am now wondering whether this accessor is even needed tbh (not really sure what use the current time is to a caller). Do we communicate internal members by prefixing with _ ?

Reply:
Internal private variables should start with _. You can define internal properties but I think if the property only has a getter or a setter and not both it might be better to define a function with just that e.g getTime()

Reply:
Accessors that only have `get` for a read-only property are fine

Reply:
I've updated members to all be internal and removed the errant property get.